sky/skylet/attempt_skylet.py

```diff
@@ -0,0 +1,23 @@
+"""Restarts skylet if version does not match"""
```

Comment:
It might be better to modify the `skylet.py` automatically check if there is an older version skylet running. If the original skylet is up-to-date, exit the current program immediately. Otherwise, kill the old skylet process and make itself as the skylet process.

So everywhere we can just call `python3 -m sky.skylet.skylet 2>&1 >> ~/.sky/skylet.log &`

```
# skylet.py
def start_skylet():
    # The previous loop
    when ...
         

if __name__ == '__main__':
    skylet_pid = subprocess.run('ps aux | grep "sky.skylet.skylet" | grep "python3 -m" | awk '{print $2}', stdout=...)
     if not skylet_pid or skylet_versoin_from_file != constants.SKYLET_VERSION:
          # Kill the old skylet
          subprocess.run(...)
          # Save the current skylet
          f.write(constants.SKYLET_VERSION)
          # Restart skylet in the current program
          start_skylet()
```

Reply:
hmm this makes things a little annoying because:
1. you have to make sure that the "existing" skylet process isn't the process that's currently running
2. when killing the old skylet, you have to find the running skylet that's not the currently running process

it doesn't seem like you can just `grep -v os.getpid()` either since the RSS or VSZ outputs of `ps aux` might contain these.

Reply:
That makes sense! Thanks for the clarification. In that case, the current way should be good enough.